api/grpcserver/grpcserver_test.go

```diff
@@ -496,13 +493,17 @@ func launchServer(tb testing.TB, cfg Config, services ...ServiceAPI) func() {
 		require.NoError(tb, err)
 	}
 
-	return func() {
-		require.NoError(tb, jsonService.Shutdown(context.Background()))
-		_ = grpcService.Close()
+	// update config with bound addresses
+	cfg.PublicListener = grpcService.BoundAddress
+	cfg.JSONListener = jsonService.BoundAddress
+
+	return cfg, func() {
+		assert.NoError(tb, jsonService.Shutdown(context.Background()))
+		assert.NoError(tb, grpcService.Close())
 	}
 }
 
-func callEndpoint(t *testing.T, endpoint, payload string) (string, int) {
+func callEndpoint(t *testing.T, cfg Config, endpoint, payload string) (string, int) {
 	url := fmt.Sprintf("http://%s/%s", cfg.JSONListener, endpoint)
```

Comment:
```suggestion
func callEndpoint(t *testing.T, address String, endpoint, payload string) (string, int) {
	url := fmt.Sprintf("http://%s/%s", address, endpoint)
```

Reply:
I went for a slightly different approach that passes `url` directly.